### ISSUE TITLE
react-native 0.59 Some language wrapped texts are cut off on Android

### ISSUE THREAD

**Issue by soh335** [facebook/react-native#25297]
React Native version:

```
  React Native Environment Info:
    System:
      OS: macOS 10.14.5
      CPU: (8) x64 Intel(R) Core(TM) i7-8559U CPU @ 2.70GHz
      Memory: 550.93 MB / 16.00 GB
      Shell: 5.5.1 - /usr/local/bin/zsh
    Binaries:
      Node: 10.10.0 - /usr/local/bin/node
      Yarn: 1.9.4 - /usr/local/bin/yarn
      npm: 6.4.1 - /usr/local/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    SDKs:
      iOS SDK:
        Platforms: iOS 12.2, macOS 10.14, tvOS 12.2, watchOS 5.2
      Android SDK:
        API Levels: 23, 25, 26, 27, 28
        Build Tools: 23.0.1, 27.0.3, 28.0.1, 28.0.3
        System Images: android-23 | Intel x86 Atom_64, android-23 | Google APIs Intel x86 Atom_64, android-28 | Intel x86 Atom_64, android-28 | Google APIs Intel x86 Atom, android-28 | Google Play Intel x86 Atom
    IDEs:
      Android Studio: 3.4 AI-183.6156.11.34.5522156
      Xcode: 10.2.1/10E1001 - /usr/bin/xcodebuild
    npmPackages:
      react: 16.8.3 => 16.8.3
      react-native: 0.59.8 => 0.59.8
```

## Steps To Reproduce

1. Add text component with wrapped text (Japanese or Chinese, not Enlgish).
2. Wrapped texts are cut off.

<img width="672" alt="スクリーンショット 2019-06-18 18 55 06" src="https://user-images.githubusercontent.com/31937/59672869-4875a880-91fb-11e9-8dcd-b8a229c7f3fb.png">

Describe what you expected to happen:

Wrapped texts are not cut off.

This problem may be related to https://github.com/facebook/react-native/issues/24837. But happen not only RTL language.

After changing to target sdk 27, this problem is not happen.(https://github.com/facebook/react-native/issues/24837#issuecomment-492061717) 

<img width="672" alt="スクリーンショット 2019-06-18 18 54 38" src="https://user-images.githubusercontent.com/31937/59673600-82937a00-91fc-11e9-97f9-00062e92f260.png">

Snack, code example, or link to a repository:

- https://snack.expo.io/@soh335/curious-beef-jerky
- https://github.com/soh335/RNWrappedTextBug

**Comment by linjson** [comment]
see Duplicate of #25275

**Comment by soh335** [comment]
@linjson thank you ! I confirmed to fix this problem after applying your patch.

```diff
--- node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/views/text/ReactTextShadowNode.java 2019-06-19 14:17:35.000000000 +0900
+++ node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/views/text/ReactTextShadowNode.java.patched 2019-06-19 14:17:30.000000000 +0900
@@ -110,6 +110,9 @@ public class ReactTextShadowNode extends
               if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
                 builder.setJustificationMode(mJustificationMode);
               }
+              if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+                  builder.setUseLineSpacingFromFallbacks(true);
+              }
               layout = builder.build();
             }

@@ -134,14 +137,18 @@ public class ReactTextShadowNode extends
                   new StaticLayout(
                       text, textPaint, (int) width, alignment, 1.f, 0.f, mIncludeFontPadding);
             } else {
-              layout =
+              StaticLayout.Builder builder =
                   StaticLayout.Builder.obtain(text, 0, text.length(), textPaint, (int) width)
                       .setAlignment(alignment)
                       .setLineSpacing(0.f, 1.f)
                       .setIncludePad(mIncludeFontPadding)
                       .setBreakStrategy(mTextBreakStrategy)
-                      .setHyphenationFrequency(Layout.HYPHENATION_FREQUENCY_NORMAL)
-                      .build();
+                      .setHyphenationFrequency(Layout.HYPHENATION_FREQUENCY_NORMAL);
+
+              if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+                  builder.setUseLineSpacingFromFallbacks(true);
+              }
+              layout = builder.build();
             }
           }
```

**Comment by soh335** [comment]
confirm to fixed at 0.60.0
